### ISSUE TITLE
[a11y] Unmapped columns not able to be mapped

### ISSUE THREAD

**Issue by grantfitzsimmons** [specify/specify7#1343]
**Describe the bug**
Unmapped column is not able to be mapped.

**To Reproduce**
1. Go to westvirginia
2. Attempt to map "Cultivated"
3. See error

**Screenshots**
![msedge_TLq9xCBdjv](https://user-images.githubusercontent.com/37256050/165789578-f2f84dcc-335c-45fc-84da-449ba538b65d.gif)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Edge

**Comment by grantfitzsimmons** [comment]
![msedge_ubU8Meh8BP](https://user-images.githubusercontent.com/37256050/165791929-64315923-11c4-4917-8fec-e68e92ef266e.gif)

New import, new field problem

**Comment by maxxxxxdlp** [comment]
Unable to replicate it, but this bug is quite serious, so please do more testing if possible.
For reference, here are the conditions for when the button is enabled (all of them have to be satisfied):
1. Data set is not uploaded
2. User has edit access to the data set
3. There is at least one column in the data set (new datasets don't have any columns)
4. Currently focused item in the Mapping Editor forms a complete path (i.e, there is a field selected in the last box)
5. The field is not already mapped to any header

The bug is likely happening in 4 or 5.

**Comment by maxxxxxdlp** [comment]
@grantfitzsimmons Thanks for helping me trace this issue
Turns out it was a subtle bug that affected several places in the WorkBench and beyond.
The core of the issue is when it checks whether a mapping path starts with given sub path, it firstly converts all paths to strings (by joining the arrays with `.`) and then using `String.prototype.startsWith` on them.
That lead to these two mapping paths being considered identical:
```
$Species.taxonAttribute.text1
$Species.taxonAttribute.text10
```
Fixed in ea829f05
